### PR TITLE
Remove loop from payout manager

### PR DESCRIPTION
**Pull request by jburley-nlok**

### CODE REVIEW

File: src/Miningcore/Payments/PayoutManager.cs [hunk 1]
```diff
@@ -63,67 +62,81 @@ public class PayoutManager
...
-                        try
-                        {
-                            var handler = await ResolveAndConfigurePayoutHandlerAsync(pool);
+        public void ScheduleNextPayout()
+        {
+            if (!cts.IsCancellationRequested)
+            {
+                // First schedule the next payout
+                logger.Info(() => $"Scheduled the next payout in {interval.TotalSeconds} seconds");
+                timer = new Timer((state) => ScheduleNextPayout(), null, (int)interval.TotalMilliseconds, System.Threading.Timeout.Infinite);
```
Comment (joel-nlok) [inline src/Miningcore/Payments/PayoutManager.cs]: so if the payout run takes longer than the payout cycle, we could have 2 running at the same time, right?

Comment (jburley-nlok) [reply src/Miningcore/Payments/PayoutManager.cs]: Addressed per our conversation

File: src/Miningcore/Payments/PayoutManager.cs [hunk 2]
```diff
@@ -62,68 +62,86 @@ public class PayoutManager
...
+            PayoutCallback();
+        }
 
-                            // resolve payout scheme
-                            var scheme = ctx.ResolveKeyed<IPayoutScheme>(pool.PaymentProcessing.PayoutScheme);
+        public void PayoutCallback() 
+        {
+            if (!cts.IsCancellationRequested)
+            {
+                if (null == payoutThread || !payoutThread.IsAlive)
```
Comment (kabilanvk) [inline src/Miningcore/Payments/PayoutManager.cs:83]: We are moving away from Task to Thread as opposed to other loops and base code. Just want to highlight so we don't fight against thread handling after this.

File: src/Miningcore/Payments/PayoutManager.cs [hunk 3]
```diff
@@ -62,68 +62,86 @@ public class PayoutManager
...
+                {
+                    // Spin up a separate thread to do the current payout
+                    logger.Info(() => "Creating a thread to process the current payouts");
+                    payoutThread = new Thread(() => ProcessPayout());
```
Comment (kabilanvk) [inline src/Miningcore/Payments/PayoutManager.cs:87]: We are moving away from Task to Thread as opposed to other loops and base code. Just want to highlight so we don't fight against thread handling after this.

**Review by kabilanvk — APPROVED** [review]

**Review by joel-nlok — APPROVED** [review]

**Pull request merged**